tests/EventsTest.cpp

```diff
@@ -44,4 +71,417 @@ TEST_CASE("Weight change")
+        {
+          pb.remove(tasks[i].get());
+          added[i] = false;
+        }
+        else
+        {
+          pb.add(tasks[i]);
+          added[i] = true;
+        }
+
+        if (added[j])
+        {
+          pb.remove(tasks[j].get());
+          added[j] = false;
+        }
+        else
+        {
+          pb.add(tasks[j]);
+          added[j] = true;
+        }
+
+        if (added[k])
+        {
+          pb.remove(tasks[k].get());
+          added[k] = false;
+        }
+        else
+        {
+          pb.add(tasks[k]);
+          added[k] = true;
+        }
+
+        // Create a gound truth problem
+        LinearizedControlProblem pbGroundTruth;
+        buildPb(pbGroundTruth, tasks, added);
+
+        // Now we solve both problem. Only pb is performing an update
+#if defined(TVM_USE_LSSOL)
+        FAST_CHECK_UNARY(solverLssol.solve(pbGroundTruth));
+#elif defined(TVM_USE_QLD)
+        FAST_CHECK_UNARY(solverQLD.solve(pbGroundTruth));
+#elif defined(TVM_USE_QUADPROG)
+        FAST_CHECK_UNARY(solverQuadprog.solve(pbGroundTruth));
+#endif
+        VectorXd s0 = pbGroundTruth.variables().value();
+
+#ifdef TVM_USE_LSSOL
+        FAST_CHECK_UNARY(solverLssol.solve(pb));
+        checkSolution(tasks, added, pbGroundTruth, s0, pb, pb.variables().value());
+#endif
+#ifdef TVM_USE_QLD
+        FAST_CHECK_UNARY(solverQLD.solve(pb));
+        checkSolution(tasks, added, pbGroundTruth, s0, pb, pb.variables().value());
+#endif
+#ifdef TVM_USE_QUADPROG
+        FAST_CHECK_UNARY(solverQuadprog.solve(pb));
+        checkSolution(tasks, added, pbGroundTruth, s0, pb, pb.variables().value());
+#endif
+      }
+    }
+  }
+}
+
+
+TEST_CASE("Systematic add/remove of one task")
+{
+  // This test is creating a large number of problems, leading to a large log.
+  // So we disable the logs here.
+  tvm::graph::internal::Logger::logger().disable();
+
+  std::array<bool, 12> added = { false, false, false, false, false, false, false, false, false, false, false, false };
+  // all combinations of true/false for added
+  for (int i = 1; i < 4096; ++i)
+  {
+    // Take the next value of added
+    for (int j = 0; j < 12; ++j)
+    {
+      added[j] = !added[j];
+      if (added[j]) break;
+    }
+
+    if (!skip(added))
+    {
+      test1Change(added);
+    }
+  }
+}
+
+TEST_CASE("Systematic add/remove of one task with substitution")
+{
+  // This test is creating a large number of problems, leading to a large log.
+  // So we disable the logs here.
+  tvm::graph::internal::Logger::logger().disable();
+
+  std::array<bool, 12> added = { true, false, false, false, false, false, false, false, false, false, false, false };
+  // all combinations of true/false for added[1..10]
+  for (int i = 1; i < 1024; ++i)
+  {
+    // Take the next value of added
+    for (int j = 1; j < 11; ++j)
+    {
+      added[j] = !added[j];
+      if (added[j]) break;
+    }
+
+    if (!skip(added))
+    {
+      test1Change(added, true);
+    }
+  }
+}
```

Comment:
Similarly:
```c++
std::bitset<12> b(0);
for(uint16_t i = 0; i < (1 << 10); ++i)
{
  b = (1 + 2 * i + (1 << 11));
  if(!skip(b))
  {
    test1Change(b, true);
  }
}
```

Reply:
Changed in 0292f935a9559b21b19e8682ca166c3388d2647a